cmd/api/main.go

```diff
@@ -279,16 +258,13 @@ func main() {
 			log.Error().Err(err).Msg("closing user store")
 		}
 
-		if err := metricsStore.Close(); err != nil {
-			log.Error().Err(err).Msg("closing metrics store")
-		}
-
-		if config.TelemetryPublisher.Enabled {
-			metricsPublisher.Close()
-		}
 		if err := executorsDB.Close(); err != nil {
 			log.Error().Err(err).Msg("closing executors db")
 		}
+
+		if err := closeTelemetryModule(ctx); err != nil {
+			log.Error().Err(err).Msg("closing telemetry module")
+		}
```

Comment:
The logic of `cli.HandleInterrupt(...)` is indirectly simplified.

Reply:
nice